aten/src/ATen/native/quantized/cuda/affine_quantizer.cu

```diff
@@ -115,7 +115,73 @@ void dequantize_tensor_per_channel_affine_cuda(
         .add_input(native::_unsafe_view(zero_points, expected_shape))
         .build();
 
-      gpu_kernel(iter, [=] GPU_LAMBDA(scalar_t value, float scale, int64_t zero_point) -> float {
+      gpu_kernel(iter, [=] GPU_LAMBDA(scalar_t value, double scale, int64_t zero_point) -> float {
+        return static_cast<float>(value.val_ - zero_point) * scale;
+      });
+    });
+}
+
+void quantize_tensor_per_channel_float_qparams_cuda(
+  Tensor rtensor,
+  Tensor qtensor,
+  Tensor scales,
+  Tensor zero_points,
+  int64_t axis) {
+
+  std::vector<int64_t> expected_shape(rtensor.dim(), 1);
+  expected_shape[axis] = rtensor.size(axis);
+
+  auto iter = TensorIteratorConfig()
+  .check_all_same_dtype(false)
+  .add_output(qtensor)
+  .add_input(rtensor)
+  .add_input(qtensor)
+  .add_input(native::_unsafe_view(scales, expected_shape))
+  .add_input(native::_unsafe_view(zero_points, expected_shape))
+  .build();
+
+
+
+  AT_DISPATCH_QINT_TYPES(
+    qtensor.scalar_type(), "quantize_tensor_per_channel_float_qparams_cuda", [&]() {
+      constexpr int64_t qmin = std::numeric_limits<underlying_t>::min();
+      constexpr int64_t qmax = std::numeric_limits<underlying_t>::max();
+      // trying to match _quantize_per_channel_ref_nd in test_quantized_tensor.py
+      gpu_kernel(
+          iter,
+          [=] GPU_LAMBDA(float raw_val, scalar_t quantized_val, float scale, float zero_point) -> scalar_t {
+            float inv_scale = 1.0f/scale;
+            int64_t qvalue =
+                static_cast<int64_t>(nearbyint(raw_val*inv_scale + zero_point));
```

Comment:
trying to match to https://github.com/pytorch/pytorch/blob/470cd6451400d9c96d20b91e3915172abc056455/test/quantization/test_quantized_tensor.py#L341-L347